packages/fast-element/src/attributes.ts

```diff
@@ -1,21 +1,139 @@
+import { Observable } from "./observation/observable";
+
+export interface ValueConverter {
+    toView(value: any): string;
+    fromView(value: string): any;
+}
+
+export type AttributeMode = "reflect" | "boolean" | "none";
+
 export type AttributeConfiguration = {
     property: string;
     attribute?: string;
+    mode?: AttributeMode;
+    converter?: ValueConverter;
 };
 
 export type DecoratorAttributeConfiguration = Omit<AttributeConfiguration, "property">;
 
+export const booleanConverter: ValueConverter = {
+    toView(value: any): string {
+        return value ? "true" : "false";
+    },
+
+    fromView(value: any): any {

```

Comment:
HTML5 allows for boolean attrbiutes to be set to the *attribute name*: https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#boolean-attributes

eg `disabled="disabled"` 

Does this converter cover that case?

Reply:
No it doesn't cover that scenario. Is there some advantage to doing it that way?

Reply:
Well, technically, the way the converter is implemented, it will interpret any string besides "false"  and empty string as being a true value. Is that ok?

Reply:
Well, if we followed what could be true, empty string would actually be true. I'm not sure how much we mind but `<input disabled />` is `<input disabled="disabled" />` is `<input disabled="" />`.

Reply:
Merged this. If there's a big issue with the above behavior, I can make a future change.

Reply:
Currently, empty string will result in false in the way the converter is implemented.

Reply:
I don't think there is an advantage to setting `disabled="disabled"` - was more commenting because it is a native language feature that we might want to mimic.

Here is a little pen of how button behaves: https://codepen.io/rice119/pen/oNXwqpP

It seems if the attribute exists then the IDL is true, otherwise it is false.

Reply:
Hmm. Minus the empty string scenario, our implementation works just like this. I need to play with how web components report the attribute changed value when the attribute is removed. If it is reported as an empty string, for example, then that would cause us a problem in terms of matching implementation exactly. If it returns null or undefined, then we can make this match with no trouble. I've got another PR coming with the docs I forgot to add with this. If I can make this change, I'll include it in that PR.

Reply:
awesome thanks @EisenbergEffect!

Reply:
Ok, I can fix this 😄 New PR incoming with the fix and the docs. Thanks for digging in on that and catching the inconsistency!